### PR TITLE
feat: Add 3cx as video channel

### PR DESCRIPTION
**Pull request by MeghalBisht**
from meeting notes

#### Short description of what this resolves:

**There is no prefix for 3cx web meetings so this is more like a custom option as I can even add any other url and it would work fine.**

Screenshot from 2021-04-17 00-09-19
Screenshot from 2021-04-17 00-10-24

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/forms/events/view/videoroom-form.hbs [hunk 1]
```diff
@@ -62,6 +62,11 @@
           @site="vimeo"
           @value={{this.data.stream.url}}
           @onChange={{action (mut this.data.stream.url)}} />
+      {{else if (eq this.data.stream.videoChannel.provider '3cx')}}
+        <Widgets::Forms::LinkField
+          @inputId="url"
+          @value={{this.data.stream.url}}
+          @onChange={{action (mut this.data.stream.url)}} />
```
Comment (iamareebjamal) [inline app/components/forms/events/view/videoroom-form.hbs:69]: Isn't it same as else?

Comment (MeghalBisht) [reply app/components/forms/events/view/videoroom-form.hbs:69]: Yes, will fix it. 
But is this correct? Why do we need another channel when it is similar to custom?

File: app/components/public/stream/video-stream.ts [hunk 2]
```diff
@@ -50,6 +53,9 @@ export default class PublicStreamVideoStream extends Component<Args> {
 
     if (provider === 'jitsi') {
       this.loading = false;
+    } else if (provider === '3cx') {
+      this.cxUrl = stream.url;
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.ts:57]: Use iframeUrl

File: app/components/public/stream/video-stream.hbs [hunk 3]
```diff
@@ -14,6 +14,9 @@
   {{#if (and (eq @videoStream.videoChannel.provider 'vimeo') this.vimeoId)}}
     <iframe src="https://player.vimeo.com/video/{{this.vimeoId}}?autoplay={{if @videoStream.extra.autoplay 1 0}}&loop={{if @videoStream.extra.loop 1 0}}" width="100%" height="100%" frameborder="0" title="Vimeo Live Stream" allow="autoplay" allowfullscreen="true"></iframe>
   {{/if}}
+  {{#if (and (eq @videoStream.videoChannel.provider '3cx') this.cxUrl)}}
```
Comment (iamareebjamal) [inline app/components/public/stream/video-stream.hbs:17]: iframeUrl case should be reused here

File: app/components/forms/events/view/videoroom-form.hbs (hunk 1 above)
Comment (iamareebjamal) [reply app/components/forms/events/view/videoroom-form.hbs:69]: Custom won't show in iframe

**Pull request merged**